### ISSUE TITLE
[Build Tools] Test Update 1

### ISSUE THREAD

**Issue by pramodya1994** [ballerina-platform/ballerina-lang#35880]
**Description:**
Need to test the following features added to update 1.

- [x] https://github.com/ballerina-platform/ballerina-lang/issues/35098
- [x] https://github.com/ballerina-platform/ballerina-lang/issues/35219
- [x] https://github.com/ballerina-platform/ballerina-lang/issues/32767
- [x] https://github.com/ballerina-platform/ballerina-lang/issues/35550
- [x] #34939
- [x] https://github.com/ballerina-platform/ballerina-lang/issues/34819
- [x] https://github.com/ballerina-platform/ballerina-lang/issues/34221
- [x] https://github.com/ballerina-platform/ballerina-lang/issues/34942
- [x] https://github.com/ballerina-platform/ballerina-lang/issues/35540
- [x] https://github.com/ballerina-platform/ballerina-lang/issues/35413
- [x] Object and function mocking
- [ ] https://github.com/ballerina-platform/ballerina-lang/pull/36224
- [ ] https://github.com/ballerina-platform/ballerina-lang/issues/36245

**Describe your task(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by Ibaqu** [comment]
Tested https://github.com/ballerina-platform/ballerina-lang/issues/35098 on pack `2201.0.1-rc1.6` 
- imported `ballerina/io` ~ ver `1.2.2` gets resolved automatically
- Deleted `Dependencies.toml` and build file
- Specified `ballerina/io` ver `1.2.0` in `Ballerina.toml` as a dependency
- Build again  ~ ver `1.2.0` gets resolved successfully

Also tried ver `1.0.0` but `1.0.1` gets resolved instead.

**Comment by Ibaqu** [comment]
Tested https://github.com/ballerina-platform/ballerina-lang/issues/35540 

~ Checked for a project and a package 
~ Pack command 
     ~ works for a package with only a compiler plugin
     ~ throws error if completely empty  
~ Build command
     ~ throws an error regardless of wether a compiler plugin exists or not

**Comment by Ibaqu** [comment]
Tested [Fail fast on errors](https://github.com/ballerina-platform/ballerina-lang/issues/35219)

Debugged to the point to confirm wether its actually failing fast or not.

**Comment by Ibaqu** [comment]
Tested [Module name is not valid with project build](https://github.com/ballerina-platform/ballerina-lang/issues/35550)

- Tested with a number as the name 
- Tested with invalid characters 

Both get changed with the new error message to check the toml. `app` gets added in front of the number

**Comment by Ibaqu** [comment]
Tested [LS getting stopped when incompatible versions are included](https://github.com/ballerina-platform/ballerina-lang/issues/34221)

- Error shows up in `Problems` section saying incompatible
- LS continues to work

**Comment by pramodya1994** [comment]
Tested
- https://github.com/ballerina-platform/ballerina-lang/issues/34819
Created package `pramodya/foo:1.0.0` and pushed it to central
Added some changes to `pramodya/foo:1.0.0` and pushed it to local repo
Used packaged `foo` in package `bar` and added the following to the `Ballerina.toml` file of package `bar`
```toml
[[dependency]]
org = "pramodya"
name = "foo"
version = "2.0.0"
repository="local"
```
Could notice following warning in the `Ballerina.toml` `[[dependency]]` section
<img width="800" alt="Screenshot 2022-05-19 at 16 30 30" src="https://user-images.githubusercontent.com/19742678/169278543-e8c48b6c-e0df-4cb2-ba25-45e8918894dc.png">

**Comment by pramodya1994** [comment]
Tested
- https://github.com/ballerina-platform/ballerina-lang/issues/34939
Ran `bal pull ballerinax/redis:2.2.1` and killed the process (`Ctrl + C`) while it's getting downloaded.
```
ballerinax/redis:2.2.1 [central.ballerina.io -> home repo]   15% [====================>                                                                                                                  ] 1514/9878 KB (0:00:05 / 0:00:27)
```
Created a new project which uses Redis package and built it without any issue.
```
$ bal build
Compiling source
	pramodya/redis_main:0.1.0
ballerinax/redis:2.2.1 [central.ballerina.io -> home repo]  100% [=====================================================================================================================================] 10951/10951 KB (0:00:11 / 0:00:00)
	ballerinax/redis:2.2.1 pulled from central successfully

Generating executable
	target/bin/redis_main.jar
```

**Comment by pramodya1994** [comment]
Tested
- https://github.com/ballerina-platform/ballerina-lang/issues/34942
Created an application package with the following `main.bal`
```bal
import ballerina/io;
import ballerinax/redis as _;
import ballerina/http as _;

public function main() {
    io:println("Hello, World!");
}
```
Built package 10 times, got the following conflicting jars warning every time.
```
warning: Detected conflicting jar files:
        'netty-buffer-4.1.71.Final.jar' dependency of 'ballerinax/redis' conflict with 'netty-buffer-4.1.77.Final.jar' dependency of 'ballerina/http'
	'netty-codec-4.1.71.Final.jar' dependency of 'ballerinax/redis' conflict with 'netty-codec-4.1.77.Final.jar' dependency of 'ballerina/http'
	'netty-common-4.1.71.Final.jar' dependency of 'ballerinax/redis' conflict with 'netty-common-4.1.77.Final.jar' dependency of 'ballerina/http'
	'netty-handler-4.1.71.Final.jar' dependency of 'ballerinax/redis' conflict with 'netty-handler-4.1.77.Final.jar' dependency of 'ballerina/http'
	'netty-resolver-4.1.71.Final.jar' dependency of 'ballerinax/redis' conflict with 'netty-resolver-4.1.77.Final.jar' dependency of 'ballerina/http'
	'netty-transport-4.1.71.Final.jar' dependency of 'ballerinax/redis' conflict with 'netty-transport-4.1.77.Final.jar' dependency of 'ballerina/http'
```

**Comment by Ibaqu** [comment]
> 

Tested this on 2201.1.1, works but now when we specify `io:1.0.0`, it resolves the version correct unlike last time when it  resolved to `io:1.0.1`.
